### DISCUSSION TITLE
rename commit with editor without commit hook

### DISCUSSION THREAD

**Discussion by tg0h** [jesseduffield/lazygit#2810] [General]
hello,

I want to rename a commit with my editor without firing the commit hooks in my repo.

currently, you can reword the commit in your editor with R in the commits panel but this fires the commit hook.

I guess I need something like a custom command that calls 'git commit --allow-empty --amend --no-verify' and opens the editor in lazygit. Is this possible?

**Comment by guglielmobartelloni** [comment]
Same I need to allow-empty commits because my organizations uses this commits to trigger CI

**Comment by mikavilpas** [comment]
I'm trying something like this out https://github.com/mikavilpas/dotfiles/commit/43a59c0eb06500f21135837a78317a7b0e913a10

disclaimer: just tried it out once, so not much real world testing on this one yet 😄 

```diff
commit 43a59c0eb06500f21135837a78317a7b0e913a10
Author: Mika Vilpas <mika.vilpas@gmail.com>
Date:   Fri May 17 16:14:10 2024 +0300

    feat(lazygit): skip re-verifying when rewording (`R`) commit in editor

diff --git a/.config/lazygit/config.yml b/.config/lazygit/config.yml
index 013db32..e3225f4 100644
--- a/.config/lazygit/config.yml
+++ b/.config/lazygit/config.yml
@@ -72,6 +72,10 @@ customCommands:
     # want to reduce the repetition in the commit messages
     command:
       "git commit --amend --no-verify --message '{{.SelectedLocalCommit.Name}}'"
+  - key: R
+    context: commits
+    command: git commit --allow-empty --amend --no-verify
+    subprocess: true
 
 keybinding:
   universal:
```
